File: pset6/credit/credit.py
from sys import exit

while True:
    try:
        ccInput = int(input("Number: "))
        if ccInput > 0:
            break
    except:
        continue

ln = (len(str(ccInput)))

if (ln is not 13) and (ln is not 15) and (ln is not 16):
    print("INVALID\n", end="")
    exit(1)

ccDigits = [];
for i in range(ln):
    ccDigits.append(ccInput % 10)
    ccInput = ccInput // 10
    i += i

sum1 = 0
for i in range(1, ln, 2):
    sumTemp = ccDigits[i] * 2
    if sumTemp >= 10:
        d1 = sumTemp // 10
        d2 = sumTemp % 10
        sumTemp = d1 + d2
    sum1 = sum1 + sumTemp

sum2 = 0
for i in range(0, ln, 2):
    sum2 = sum2 + ccDigits[i]
sumFinal = sum1 + sum2

if (sumFinal % 10 is not 0):
    print("INVALID\n", end="")
else:
    if ((ln == 13 or ln == 16) and (ccDigits[12] == 4 or ccDigits[15] == 4)):
        print("VISA\n", end="")
    elif ((ln == 15 and ccDigits[14] == 3) and (ccDigits[13] == 4, 7)):
        print("AMEX\n", end="")
    elif ((ln == 16 and ccDigits[15] == 5) and (ccDigits[14] == 1, 2, 3, 4, 5)):
        print("MASTERCARD\n", end="")
    else:
        print("INVALID\n", end="")

File: pset1/pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    long ccInput;
    do
    {
        ccInput = get_long("Credit Card Number: \n");
    }
    while (ccInput < 0);

    long ccNumber = ccInput;
    int digits = 0;

    while (ccInput > 0)
    {
        ccInput /= 10;
        digits++;
    }

    if (digits != 13 && digits != 15 && digits != 16)
    {
        printf("INVALID\n");
    }
    else
    {
        
        int ccDigits[digits];
        for (int i = 0; i < digits; i++)
        {
            ccDigits[i] = ccNumber % 10;
            ccNumber /= 10;
        }
    
        int d1 = 0;
        int d2 = 0;
        int sumTemp = 0;
        int sum1 = 0;
        for (int i = 1; i < digits; i = i + 2)
        {
            sumTemp = (ccDigits[i] * 2);
            if (sumTemp >= 10)
            {
                d1 = sumTemp / 10;
                d2 = sumTemp % 10;
                sumTemp = d1 + d2;
            }
            sum1 = sum1 + sumTemp;
        }
    
        int sum2 = 0;
        for (int i = 0; i < digits; i = i + 2)
        {
            sum2 = sum2 + ccDigits[i];
        }
    
        int sumfinal = sum1 + sum2;
        if (sumfinal % 10 != 0)
        {
            printf("INVALID\n");
        }
        else
        {
            if ((digits == 13 || digits == 16) && (ccDigits[12] == 4 || ccDigits[15] == 4))
            {
                printf("VISA\n");
            }
            else if ((digits == 15 && ccDigits[14] == 3) && (ccDigits[13] == 4 || ccDigits[13] == 7))
            {
                printf("AMEX\n");
            }
            else if ((digits == 16 && ccDigits[15] == 5) && 
                     (ccDigits[14] == 1 || ccDigits[14] == 2 || ccDigits[14] == 3 || ccDigits[14] == 4 || ccDigits[14] == 5))
            {
                printf("MASTERCARD\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
    }
}


